install-binary.sh

```diff
@@ -1,95 +1,17 @@
-
-: ${HELM_PLUGIN_PATH:="$(helm home)/plugins/helm-unittest"}
-
-# Convert the HELM_PLUGIN_PATH to unix if cygpath is
-# available. This is the case when using MSYS2 or Cygwin
-# on Windows where helm returns a Windows path but we
-# need a Unix path
-if type cygpath &> /dev/null; then
-  HELM_PLUGIN_PATH=$(cygpath -u $HELM_PLUGIN_PATH)
-fi
-
-if [[ $SKIP_BIN_INSTALL == "1" ]]; then
-  echo "Skipping binary install"
-  exit
-fi
-
-# initArch discovers the architecture for this system.
-initArch() {
-  ARCH=$(uname -m)
-  case $ARCH in
-    armv5*) ARCH="armv5";;
-    armv6*) ARCH="armv6";;
-    armv7*) ARCH="armv7";;
-    aarch64) ARCH="arm64";;
-    x86) ARCH="386";;
-    x86_64) ARCH="amd64";;
-    i686) ARCH="386";;
-    i386) ARCH="386";;
-  esac
-}
-
-# initOS discovers the operating system for this system.
-initOS() {
-  OS=$(echo `uname`|tr '[:upper:]' '[:lower:]')
-
-  case "$OS" in
-    # Msys support
-    msys*) OS='windows';;
-    # Minimalist GNU for Windows
-    mingw*) OS='windows';;
-    darwin) OS='macos';;
-  esac
-}
-
-# verifySupported checks that the os/arch combination is supported for
-# binary builds.
-verifySupported() {
-  local supported="linux-arm64\linux-amd64\nmacos-amd64\nwindows-amd64"
-  if ! echo "${supported}" | grep -q "${OS}-${ARCH}"; then
-    echo "No prebuild binary for ${OS}-${ARCH}."
-    exit 1
-  fi
-
-  if ! type "curl" >/dev/null 2>&1 && ! type "wget" >/dev/null 2>&1; then
-    echo "Either curl or wget is required"
-    exit 1
-  fi
-}
-
 
-# getDownloadURL checks the latest available version.
-getDownloadURL() {
-  local version=$(git -C $HELM_PLUGIN_PATH describe --tags --exact-match 2>/dev/null)
-  if [ -n "$version" ]; then
-    DOWNLOAD_URL="https://github.com/$PROJECT_GH/releases/download/$version/$PROJECT_NAME-$OS-$ARCH.tgz"
-  else
-    # Use the GitHub API to find the download url for this project.
-    local url="https://api.github.com/repos/$PROJECT_GH/releases/latest"
-    if type "curl" > /dev/null; then
-      DOWNLOAD_URL=$(curl -s $url | grep $OS-$ARCH | awk '/\"browser_download_url\":/{gsub( /[,\"]/,"", $2); print $2}')
-    elif type "wget" > /dev/null; then
-      DOWNLOAD_URL=$(wget -q -O - $url | grep $OS-$ARCH | awk '/\"browser_download_url\":/{gsub( /[,\"]/,"", $2); print $2}')
-    fi
-  fi
-}
-
-# downloadFile downloads the latest binary package and also the checksum
-# for that binary.
 downloadFile() {
-  PLUGIN_TMP_FILE="/tmp/${PROJECT_NAME}.tgz"
+  # Always use version from plugin.yaml for the download url
+  DOWNLOAD_URL="https://github.com/rancher/helm-unittest/releases/download/$HELM_UNITTEST_VERSION/$PROJECT_NAME-linux-$ARCH.tgz"
+  PLUGIN_TMP_FILE="/tmp/$PROJECT_NAME.tgz"
   echo "Downloading $DOWNLOAD_URL"
-  if type "curl" >/dev/null 2>&1; then
-    curl -L "$DOWNLOAD_URL" -o "$PLUGIN_TMP_FILE"
-  elif type "wget" >/dev/null 2>&1; then
-    wget -q -O "$PLUGIN_TMP_FILE" "$DOWNLOAD_URL"
-  fi
+  curl -L "$DOWNLOAD_URL" -o "$PLUGIN_TMP_FILE"
```

Comment:
may not need the L anymore if there's no redirect

Reply:
there is a redirect